src/objective-c/tests/CoreCronetEnd2EndTests/h2_ssl.m

```diff
@@ -0,0 +1,203 @@
+  GPR_ASSERT(f->client != NULL);
+}
+
+static void chttp2_init_server_secure_fullstack(
+    grpc_end2end_test_fixture *f, grpc_channel_args *server_args,
+    grpc_server_credentials *server_creds) {
+  fullstack_secure_fixture_data *ffd = f->fixture_data;
+  if (f->server) {
+    grpc_server_destroy(f->server);
+  }
+  f->server = grpc_server_create(server_args, NULL);
+  grpc_server_register_completion_queue(f->server, f->cq, NULL);
+  GPR_ASSERT(grpc_server_add_secure_http2_port(f->server, ffd->localaddr,
+                                               server_creds));
+  grpc_server_credentials_release(server_creds);
+  grpc_server_start(f->server);
+}
+
+void chttp2_tear_down_secure_fullstack(grpc_end2end_test_fixture *f) {
+  fullstack_secure_fixture_data *ffd = f->fixture_data;
+  gpr_free(ffd->localaddr);
+  gpr_free(ffd);
+}
+
+static void cronet_init_client_simple_ssl_secure_fullstack(
+    grpc_end2end_test_fixture *f, grpc_channel_args *client_args) {
+  grpc_arg ssl_name_override = {GRPC_ARG_STRING,
+                                GRPC_SSL_TARGET_NAME_OVERRIDE_ARG,
+                                {"foo.test.google.fr"}};
+  
+  grpc_channel_args *new_client_args =
+      grpc_channel_args_copy_and_add(client_args, &ssl_name_override, 1);
+  [Cronet setHttp2Enabled:YES];
+  [Cronet start];
+  cronet_engine *cronetEngine = [Cronet getGlobalEngine];
+  
+  cronet_init_client_secure_fullstack(f, new_client_args, cronetEngine);
+  grpc_channel_args_destroy(new_client_args);
+}
+
+static int fail_server_auth_check(grpc_channel_args *server_args) {
+  size_t i;
+  if (server_args == NULL) return 0;
+  for (i = 0; i < server_args->num_args; i++) {
+    if (strcmp(server_args->args[i].key, FAIL_AUTH_CHECK_SERVER_ARG_NAME) ==
+        0) {
+      return 1;
+    }
+  }
+  return 0;
+}
+
+static void chttp2_init_server_simple_ssl_secure_fullstack(
+    grpc_end2end_test_fixture *f, grpc_channel_args *server_args) {
+  grpc_ssl_pem_key_cert_pair pem_cert_key_pair = {test_server1_key,
+                                                  test_server1_cert};
+  grpc_server_credentials *ssl_creds =
+      grpc_ssl_server_credentials_create(NULL, &pem_cert_key_pair, 1, 0, NULL);
+  if (fail_server_auth_check(server_args)) {
+    grpc_auth_metadata_processor processor = {process_auth_failure, NULL, NULL};
+    grpc_server_credentials_set_auth_metadata_processor(ssl_creds, processor);
+  }
+  chttp2_init_server_secure_fullstack(f, server_args, ssl_creds);
+}
+
+/* All test configurations */
+
+static grpc_end2end_test_config configs[] = {
+    {"chttp2/simple_ssl_fullstack",
+     FEATURE_MASK_SUPPORTS_DELAYED_CONNECTION |
+         FEATURE_MASK_SUPPORTS_PER_CALL_CREDENTIALS,
+     chttp2_create_fixture_secure_fullstack,
+     cronet_init_client_simple_ssl_secure_fullstack,
+     chttp2_init_server_simple_ssl_secure_fullstack,
+     chttp2_tear_down_secure_fullstack},
+};
+
+int main(int argc, char **argv) {
```

Comment:
We try hard to keep these fixtures simple, meaning efforts to factor out
common parts are frowned upon: one should never need to debug the fixtures
to figure out which core API calls it ends up making.

On Sun, Jul 17, 2016, 4:16 PM Muxi Yan <notifications@github.com> wrote:

> In src/objective-c/tests/CoreCronetEnd2EndTests/h2_ssl.m
> <https://github.com/grpc/grpc/pull/7368#discussion_r71091897>:
>
> > +  chttp2_init_server_secure_fullstack(f, server_args, ssl_creds);
> > +}
> > +
> > +/* All test configurations */
> > +
> > +static grpc_end2end_test_config configs[] = {
> > +    {"chttp2/simple_ssl_fullstack",
> > +     FEATURE_MASK_SUPPORTS_DELAYED_CONNECTION |
> > +         FEATURE_MASK_SUPPORTS_PER_CALL_CREDENTIALS,
> > +     chttp2_create_fixture_secure_fullstack,
> > +     cronet_init_client_simple_ssl_secure_fullstack,
> > +     chttp2_init_server_simple_ssl_secure_fullstack,
> > +     chttp2_tear_down_secure_fullstack},
> > +};
> > +
> > +int main(int argc, char **argv) {
>
> In my understanding we duplicated it because it does not worth spending
> time worrying about duplication and refactoring codes for duplication in
> different test fixtures... correct me if I am wrong :) By replacing I mean
> in the future we might want to use another fixture (not h2_ssl) for the
> testing in XCode. Leaving main() inside the fixture would make the
> replacement (a little bit) easier.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grpc/grpc/pull/7368/files/e04089e7c043e34e79bd077ca9a8c0b98d6c9ec7#r71091897>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJpuddwbiYrStDSuN_GO48KmZtd1CDk9ks5qWrfXgaJpZM4JLvdh>
> .
>